File: cpp/Token.cpp
#include "Token.hpp"
#include <stdio.h>

Token::Token(int type, int line, int column, void * value) {

		if (type == 0) 
			fprintf(stderr,"null token\n");

		type = type;
		line = line;
		column = column;
		value = value;
		svalue = (char *) value;
}


int Token::get_token_id() {
	return type;
}

std::string Token::to_string(){
	/* Let us hope what is in value can be shown as a proper String */
	std::string s = "";
	s += std::to_string(type) +" "+ std::to_string(line) +" "+ std::to_string(column)+ " " + svalue;
	return s;
}
	

File: cpp/Token.rb
class Token
	#    TokenType type;
	#    int       line;
	#    int       column;
	#    byte[]    value;
	#    String    svalue;
	attr_accessor :type, :line, :column, :value, :svalue

	def initialize (type, line, column, value)
		if type == nil
			puts "token nil"
		end
		self.type = type
		self.line = line
		self.column = column
		self.value = value
		self.svalue = value
	end

	def getTokenId()
		return self.type
	end

	def to_s()
		# Let us hope what is in value can be shown as a proper String
		return self.type.to_s()
		+ " " + self.line
		+ " " + self.column
		+ " \"" + self.svalue + "\"";
	end
end
